scripts/setup-centos8.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -efx -o pipefail
+# Some of the packages must be build with the same compiler flags
+# so that some low level types are the same size. Also, disable warnings.
+SCRIPTDIR=$(dirname "${BASH_SOURCE[0]}")
+source $SCRIPTDIR/setup-helper-functions.sh
+CPU_TARGET="${CPU_TARGET:-avx}"
+NPROC=$(getconf _NPROCESSORS_ONLN)
+export CFLAGS=$(get_cxx_flags $CPU_TARGET)  # Used by LZO.
+export CXXFLAGS=$CFLAGS  # Used by boost.
+export CPPFLAGS=$CFLAGS  # Used by LZO.
+
+function dnf_install {
+  dnf install -y -q --setopt=install_weak_deps=False "$@"
+}
+
+dnf_install epel-release dnf-plugins-core # For ccache, ninja
+dnf config-manager --set-enabled powertools
+dnf_install ninja-build ccache gcc-toolset-9 git wget which libevent-devel \
+  openssl-devel re2-devel libzstd-devel lz4-devel double-conversion-devel \
+  libdwarf-devel curl-devel
+
+dnf remove -y gflags
+
+# Required for Thrift
+dnf_install autoconf automake libtool bison flex python3
+
+# install sphinx for doc gen
+pip3 install sphinx sphinx-tabs breathe sphinx_rtd_theme
+
+# Activate gcc9; enable errors on unset variables afterwards.
+source /opt/rh/gcc-toolset-9/enable || exit 1
+set -u
+
+function cmake_install {
+  cmake -B "$1-build" -GNinja -DCMAKE_CXX_STANDARD=17 \
+    -DCMAKE_CXX_FLAGS="${CFLAGS}" -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_BUILD_TYPE=Release -Wno-dev "$@"
+  ninja -C "$1-build" install
+}
+
+function wget_and_untar {
+  local URL=$1
+  local DIR=$2
+  mkdir -p "${DIR}"
+  wget -q --max-redirect 3 -O - "${URL}" | tar -xz -C "${DIR}" --strip-components=1
+}
+
+# untar cmake binary release directly to /usr.
+wget_and_untar https://github.com/Kitware/CMake/releases/download/v3.17.5/cmake-3.17.5-Linux-x86_64.tar.gz /usr &
```

Comment:
can't we just `dnf install` some of the dependencies like we do for ubuntu instead of building from source? like `cmake`, `boost-devel`, `gflags-devel`, `glog-devel`, `snappy-devel`, ... instead of building from source?

Reply:
dnf install is less powerful then apt install in Ubuntu. Say with cmake，currently，cannot install with dnf.

Reply:
I can give a try but on Arrow we install it via `dnf install`, see an example on a base centos8 docker image here:
https://github.com/apache/arrow/blob/master/dev/tasks/linux-packages/apache-arrow/yum/centos-8-stream/Dockerfile#L26-L34

Reply:
cmake version of dnf installing is OK.
Have use used "dnf install cmake"